src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -917,11 +923,74 @@ def _initialize(self) -> None:
                     raise
 
         self._alembic = Alembic(self.engine)
+
+        self.db_backup_file_path = (
+            f"{self.config.backup_directory}/{self.engine.url.database}.sql"
+        )
+
         if (
             not self.skip_migrations
             and ENV_ZENML_DISABLE_DATABASE_MIGRATION not in os.environ
         ):
-            self.migrate_database()
+            # TODO add a try catch block around this fn call
+            # perform DB restore if exception is raised.
+            try:
+                self.migrate_database()
+            except Exception as e:
+                logger.error(
+                    "Failed to migrate the database. Attempting to restore "
+                    "the database from a backup."
+                )
+                self.restore_database()
```

Comment:
better to add a try/finally around this and log the exception with `logger.exception` in case the restoration doesn't work either

Reply:
Makes sense, i've added it.